### PR TITLE
[WIP] Add Python Dev Env and add Python API for DeltaTable.

### PR DESCRIPTION
**Pull request by ZhitongDB**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -6,5 +6,25 @@ jobs:
     steps:
       - checkout
       - run:
-          name: Run tests
+          name: Prepare Python env # install python3.7, python3-pip and pipenv for pyspark.
```
Comment (zsxwing) [inline .circleci/config.yml]: Can we try to build a custom image to speed up the tests? https://circleci.com/docs/2.0/custom-images/

File: .circleci/config.yml [hunk 2]
```diff
@@ -6,5 +6,25 @@ jobs:
...
+            eval "$(pyenv init -)"
+            eval "$(pyenv virtualenv-init -)"
+            pyenv install 3.7.4
+            pyenv global system 3.7.4
+            pipenv install
+      - run:
+          name: Run Pyhton tests # build the Delta Lake and run the python tests
+          command: |
+            build/sbt "++2.11.12 package"
+            pipenv run spark-submit --jars target/scala-2.11/delta-core_2.11-0.3.1-SNAPSHOT.jar src/test/python/test_deltatable.py
```
Comment (zsxwing) [inline .circleci/config.yml]: Does `target/scala-2.11/delta-core_2.11-*-SNAPSHOT.jar` work?

Comment (ZhitongDB) [reply .circleci/config.yml]: yes, it works, the failure before was caused by wrong order of option params.

File: .circleci/config.yml (hunk 1 above)
Comment (ZhitongDB) [reply .circleci/config.yml]: on my way

File: .circleci/config.yml (hunk 2 above)
Comment (ZhitongDB) [reply .circleci/config.yml]: changed.

**Comment by zsxwing** [comment]
Closing this. This has been replaced with #169

**Pull request closed without merging**